.github/workflows/linux-build.yml

```diff
@@ -70,6 +70,16 @@ jobs:
         # it doesn't work
         run: git config --global --add safe.directory /__w/velox/velox
 
+      - name: Install cmake
+        run: python3 -m pip install cmake==3.28.3
```

Comment:
Suggestion: If it's possible to separate the CMake version bump into a smaller PR that can be reviewed/merged sooner, that would be helpful to me!

Reply:
Of course! Precisely,  submitted this PR to collect this type of feedback.
I foresee at least 3 PR resulting from this one:
- Update to cmake 3.28.
- Refactor the existing cmake modules.
- Add the dependencies of Google Cloud CPP as cmake modules.

Reply:
I don't know how to that from a regular PR.

Reply:
Replied below for this, its pretty straightforward but does require another PR.